src/main/java/gregtech/api/capability/impl/GhostCircuitItemStackHandler.java

```diff
@@ -0,0 +1,114 @@
+package gregtech.api.capability.impl;
+
+import gregtech.api.capability.INotifiableHandler;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.recipes.ingredients.IntCircuitIngredient;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.math.MathHelper;
+import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GhostCircuitItemStackHandler implements IItemHandlerModifiable, INotifiableHandler {
+
+    public static final int NO_CONFIG = -1;
+
+    private final List<MetaTileEntity> notifiableEntities = new ArrayList<>();
+
+    private int circuitConfig = NO_CONFIG;
+    private ItemStack circuitStack = ItemStack.EMPTY;
+
+    public int getCircuitConfig() {
+        return this.circuitConfig;
+    }
+
+    public boolean hasCircuitConfig() {
+        return this.circuitConfig != NO_CONFIG;
+    }
+
+    public void setCircuitConfig(int config) {
+        if (config < IntCircuitIngredient.CIRCUIT_MIN || config > IntCircuitIngredient.CIRCUIT_MAX)
+            config = NO_CONFIG;
```

Comment:
Would logging a warning / throwing an exception here be preferred? I feel like something providing an invalid config would indicate a much larger problem that we should record.

Reply:
Receiving invalid config for input is part of the method contract, at least receiving `NO_CONFIG` is. The code in question aligns config value to one invalid value for convenience.

I think this is fine to leave it as-is, but if you want this to be more pedantic then I'll change it.